compiler/resolution/nilChecking.cpp

```diff
@@ -644,6 +724,111 @@ static Expr* getSingleDefExpr(Expr* expr) {
   return NULL;
 }
 
+// SE->symbol() is known to be nil/non-nil. If it comes from a MOVE or CAST,
+// propagate to their arguments.
+// For efficiency, the caller should assure isSymbolAnalyzed(SE).
+// Returns true and skips updating 'OUT' if the branch in question is unreachable.
+static bool adjustTestArgChain(SymExpr* SE, bool isAllocated, AliasMap& OUT) {
+  SymExpr* curr = SE;
+  while (true) {
```

Comment:
I'd appreciate a comment here saying something like "loop over the definitions for temporary symbols"